### DISCUSSION TITLE
Why is the count query slower than select *?

### DISCUSSION THREAD

**Discussion by alexmlong** [supabase/supabase#20489] [Questions]
(I just posted this on StackOverflow as well: https://stackoverflow.com/questions/77827364/why-is-supabase-count-query-much-slower-than-select)

When I execute a count query using supabase's syntax, `.select("*", { count: "exact", head: true })`, it's significantly slower than a `select("*")` query. It's also significantly slower than if I just run the query in the Supabase query UI, i.e. `SELECT COUNT(*) FROM...`. Why is that? Is there someway to speed it up? I've tried using `estimated` or `planned` settings for `count` but there's no noticeable speed increase. I notice in the Chrome network inspector it doesn't actually modify the query it's sending. Screenshots from the network tab are below.

Queries in chrome, first is select *, second is count:

![image](https://github.com/supabase/supabase/assets/794661/e9fcfb68-67ca-4a0e-b1f4-f15379c4ae37)

First query payload:

![image](https://github.com/supabase/supabase/assets/794661/24a0130a-87e2-43d4-9842-ac74aa741676)

Second query payload:

![image](https://github.com/supabase/supabase/assets/794661/8279aba3-771b-496c-9240-f40181a8a2c3)

This is even the case for queries that return thousands of results. I'm trying to use the count query for pagination purposes in a search result table but given the slowness of the count query I'm actually finding it more efficient to just query for the full set of results without a limit and do the pagination on the front end which doesn't feel right...

TLDR: Querying for count, expected it to be much faster than fetching all the results, it was much slower instead

**Comment by GaryAustin1** [comment]
If you google `postgres why is count slow` or similar you will see lots of discussions on why.  But basically it has to process the entire table.

You should also look at your RLS if enabled and go thru this guide as it makes a huge difference on selects with or without count.
https://supabase.com/docs/guides/database/postgres/row-level-security#rls-performance-recommendations
